isamples_export_client/duckdb_utilities.py

```diff
@@ -0,0 +1,27 @@
+import json
+from typing import Optional
+
+import duckdb
+
+
+class GeoFeaturesResult:
+    def __init__(self, bbox: dict, geo_json: str):
+        self.geo_json_dict = json.loads(geo_json)
+        # This comes out of DuckDB as a dictionary with keys, but STAC expects an array like [min_x, min_y, max_x, max_y]
+        self.bbox = [bbox["min_x"], bbox["min_y"], bbox["max_x"], bbox["max_y"]]
+
+    def __repr__(self):
+        return f"GeoFeaturesResult geo_json={self.geo_json}, bbox={self.bbox}"
+
+
+def read_geo_features_from_jsonl(filename: str) -> Optional[GeoFeaturesResult]:
+    con = duckdb.connect()
+    con.install_extension("spatial")
+    con.load_extension("spatial")
+    con.read_json(filename, format="newline_delimited")
+    q = f"select ST_Extent(envelope)as bb, ST_AsGEOJSON(envelope) as poly from (select ST_Envelope_Agg(ST_Point(produced_by.sampling_site.longitude, produced_by.sampling_site.latitude)) as envelope from '{filename}' where produced_by.sampling_site.longitude is not null)"
```

Comment:
Minor issue here, but consider using parameters as a general practice instead of f-string for sql, https://duckdb.org/docs/api/python/dbapi#prepared-statements

Reply:
Hmmmm, I don't seem to be able to get it to work with binding the filename to a parameter.  I've tried all of the following:

```
q = "select ST_Extent(envelope)as bb, ST_AsGEOJSON(envelope) as poly from (select ST_Envelope_Agg(ST_Point(produced_by.sampling_site.longitude, produced_by.sampling_site.latitude)) as envelope from '$json_file' where produced_by.sampling_site.longitude is not null)"
    con.execute(q, {"json_file": filename})

q = "select ST_Extent(envelope)as bb, ST_AsGEOJSON(envelope) as poly from (select ST_Envelope_Agg(ST_Point(produced_by.sampling_site.longitude, produced_by.sampling_site.latitude)) as envelope from '$1' where produced_by.sampling_site.longitude is not null)"
    con.execute(q, [filename])

    q = "select ST_Extent(envelope)as bb, ST_AsGEOJSON(envelope) as poly from (select ST_Envelope_Agg(ST_Point(produced_by.sampling_site.longitude, produced_by.sampling_site.latitude)) as envelope from '?' where produced_by.sampling_site.longitude is not null)"
    con.execute(q, [filename])
```

And get various errors indicating the table named `?` or `$1` or `$json_file` doesn't exist.